blaze/expr/arithmetic.py

```diff
@@ -50,18 +50,38 @@ def name(o):
         return None
 
 
-class BinOp(ElemWise):
-    __slots__ = '_hash', 'lhs', 'rhs'
+class Op(ElemWise):
+    __slots__ = '_hash', 'operands'
+    __inputs__ = 'operands'
```

Comment:
this needs to be a `tuple`, because it's used everywhere like this

```python
map(function, [getattr(expr, inp) for inp in expr.__inputs__])
```

in your implementation that would try to call

```
getattr(expr, 'o')
getattr(expr, 'p')
...
```
and of course expressions don't have those attributes